### PR TITLE
compile test for incident field laser profiles

### LINKED ISSUE #4390: incident field implementation not covered by CI
All incident field implementations are not covered by the CI, it is not clear if all are compiling :-(

### PR DESCRIPTION
**Pull request by psychocoderHPC**
fix #4390

Test all predifined incident field laser profiles.

- [x] rebase against #4394 to pull incident field bug fix

### CODE REVIEW

File: share/picongpu/tests/compileLaser/README.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Compile Test for Incident field laser profiles
+==============================================
+
+This test compiles the yee field solver and the incident field laser profiles for two, three dimensions and 32/64bit precision.
```
Comment (PrometheusPi) [inline share/picongpu/tests/compileLaser/README.rst]:
```suggestion
This test compiles the yee field solver and the incident field laser profiles for two/three dimensions and 32/64bit precision.
```

Just a suggestion

File: share/picongpu/tests/compileLaser/include/picongpu/param/fileOutput.param [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+     */
+    using FieldTmpSolvers = MakeSeq_t<>;
+
+    /** FileOutputFields: Groups all Fields that shall be dumped *************/
+
+    /** Possible native fields: FieldE, FieldB, FieldJ
+     */
+    using NativeFileOutputFields = MakeSeq_t<>;
+
+    using FileOutputFields = MakeSeq_t<>;
```
Comment (PrometheusPi) [inline share/picongpu/tests/compileLaser/include/picongpu/param/fileOutput.param:46]: Why do you need to set the output to be empty? Couldn't one not just create no output?

File: share/picongpu/tests/compileLaser/include/picongpu/param/incidentField.param [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+            using AllLaserProfiles = MakeSeq_t<
+                profiles::DispersiveLaser<>,
+                profiles::ExpRampWithPrepulse<>,
+                profiles::Free<ZeroField, ZeroField>,
+                profiles::GaussianBeam<>,
+                profiles::PlaneWave<>,
+                profiles::Polynom<>,
+                profiles::PulseFrontTilt<>,
+                profiles::Wavepacket<>>;
```
Comment (PrometheusPi) [inline share/picongpu/tests/compileLaser/include/picongpu/param/incidentField.param:106]: Do these profiles do anything without params?

File: share/picongpu/tests/compileLaser/include/picongpu/param/isaac.param [hunk 4]
```diff
@@ -0,0 +1,79 @@
+/* Copyright 2016-2022 Alexander Matthes
```
Comment (PrometheusPi) [inline share/picongpu/tests/compileLaser/include/picongpu/param/isaac.param:1]: Can this file be removed?

**Review by PrometheusPi — CHANGES_REQUESTED** [review]
Some question and I think the ISAAC file could be removed.

File: share/picongpu/tests/compileLaser/include/picongpu/param/fileOutput.param (hunk 2 above)
Comment (PrometheusPi) [reply share/picongpu/tests/compileLaser/include/picongpu/param/fileOutput.param:46]: Discussed offline with @psychocoderHPC - to reduce compile time of the test

File: share/picongpu/tests/compileLaser/include/picongpu/param/isaac.param (hunk 4 above)
Comment (PrometheusPi) [reply share/picongpu/tests/compileLaser/include/picongpu/param/isaac.param:1]: This reduced compile time - discussed offline with @psychocoderHPC

File: share/picongpu/tests/compileLaser/include/picongpu/param/incidentField.param (hunk 3 above)
Comment (psychocoderHPC) [reply share/picongpu/tests/compileLaser/include/picongpu/param/incidentField.param:106]: Yes each profiles is shipped with default parameters. Therefore the full algorithm is compiled.

**Comment by psychocoderHPC** [comment]
> Some question and I think the ISAAC file could be removed.

ISAAC and fileOutput.param are added to reduce the compile time. our default files always build a lot of derived particle attribute solvers for IO, this is not required for this compile testing.

**Review by PrometheusPi — APPROVED** [review]

**Comment by psychocoderHPC** [comment]
can be merged!

**Pull request merged**